### PR TITLE
[WIP] LLm llm-cli chat mode

### PR DESCRIPTION
**Pull request by cyita**
## Description

LLM llm-cli supports interactive mode for Llama & GptNeoX models. 
In this mode, users can always interrupt generation by pressing Ctrl+C and entering one or more lines of text, which will be converted into tokens and appended to the current context.

#### Linux
- [x] llama
- [ ] gptneox

#### Windows
- [x] llama
- [ ] gptneox

### 1. Why the change?

LLM llm-cli supports interactive mode for Llama & GptNeoX models.

### 2. User API changes

```bash
   #chat mode
   #Note: The chat mode only support LLaMA (e.g., *vicuna*), GPT-NeoX (e.g., *redpajama*)for now.
   llm-chat -m "/path/to/output/model.bin" -x llama
```

##### llama
image

##### llama
image

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [ ] Document test
- [x] Manual test
- [ ] ...

### CODE REVIEW

**Comment by jason-dai** [comment]
Maybe we should support the "chat" mode?

**Comment by jason-dai** [comment]
> ### 2. User API changes
> ```shell
> #interactive mode
> #Note: The interactive mode only support LLaMA (e.g., *vicuna*), GPT-NeoX (e.g., *redpajama*) for now.
> llm-cli -m "/path/to/output/model.bin" -x llama -i

change to `llm-chat`

> 
> #instruction mode with Alpaca under interactive mode
> llm-cli -m "/path/to/output/model.bin" -x llama -i --ins
> ```

no need to support

**Comment by cyita** [comment]
## For redparama:
### Precondition
We should use `Chat` model for chat. For example: `togethercomputer/RedPajama-INCITE-Chat-3B-v1` and `togethercomputer/RedPajama-INCITE-Chat-7B-v1`. The Instruct model will tell with itself and never let the user say something.

### Current status
The main.cpp in llm.cpp (gptneox) cannot fully support the chat function for redpajama models. But the original redpajama.cpp has a script called redpajama-chat.cpp that can support it. So we can integrate the code into main.cpp to support the chat function. However, since Yishuo is refracting the main.cpp files in llm.cpp, chat for gptneox will be postponed. And 

### Comparison between main.cpp and redpajama-chat.cpp
Model: redpajama-chat-3b
#### main.cpp
Sometimes it will complete the user's input before generating the response.
image

image

#### redpajama-chat.cpp
image

File: python/llm/src/bigdl/llm/cli/prompts/alpaca.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Below is an instruction that describes a task. Write a response that appropriately completes the request.
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/cli/prompts/alpaca.txt:1]: not needed

File: python/llm/src/bigdl/llm/cli/prompts/chat-with-ai.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+A chat between a curious human USER and an artificial intelligence assistant ChatLLaMa. The assistant gives helpful, detailed, and polite answers to the human's questions.
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/cli/prompts/chat-with-ai.txt:1]: change to chant-with-llm.txt

File: python/llm/src/bigdl/llm/cli/prompts/chat-with-ai.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+USER: Hello, ChatLLaMa.
+ChatLLaMa: Hello.
```
Comment (jason-dai) [inline python/llm/src/bigdl/llm/cli/prompts/chat-with-ai.txt:4]: change to ChatLLM

**Review by rnwang04 — APPROVED** [review]
LGTM

**Comment by cyita** [comment]
Merge first, and the gptneox chat will be supported in the next PR.

**Pull request merged**